craft_application/commands/base.py

```diff
@@ -20,7 +20,7 @@
 
 from craft_cli import BaseCommand, emit
 
-if TYPE_CHECKING:
+if TYPE_CHECKING:  # pragma: no cover
```

Comment:
can you explain the reasoning for putting argparse and craft_application inside this if?

Reply:
It comes along with ruff's [flake8-type-checking](https://beta.ruff.rs/docs/rules/#flake8-type-checking-tch) addition.

If we don't like the type checking blocks we can disable all but TCH004 and TCH005 (which will prevent accidental use of type checking blocks). Now's a good time to move that change back up to starbase too.

Reply:
we should bring it up to the group but personally I think we should keep the `if TYPE_CHECKING` guard only to break import cycles. Worrying about performance impact (the reasoning in the link) is just premature imo.